orchestrator-salt/src/main/resources/salt/salt/postgresql/upgrade/scripts/restore_db.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# restore_db.sh
+# This script is needed as Azure does not support in place RDS upgrade
+# This script:
+#  - expects the output of a successful backup, in a specific location
+#  - loads the global objects and roles from a SQL file with `psql`,
+#     - if the file does not exist then aborts
+#     - some errors are expected to happen due to already existing objects in the target database
+#  - for each service:
+#     - drops and creates the respective database of the service
+#     - loads the backup data in a parallel manner with `pg_restore`
+#  - if all is successful, then it removes the backup
+
+set -o nounset
+set -o pipefail
+
+if [[ $# -ne 3 || "$1" == "None" ]]; then
+  echo "Invalid inputs provided"
+  echo "Script accepts 3 inputs:"
+  echo "  1. PostgreSQL host name."
+  echo "  2. PostgreSQL port."
+  echo "  3. PostgreSQL user name."
+  exit 1
+fi
+
+source /opt/salt/scripts/common_utils.sh
+
+HOST="$1"
+PORT="$2"
+USERNAME="$3"
+LOGFILE=/var/log/postgres_upgrade_restore.log
+echo "Logs at ${LOGFILE}"
+
+exec 3>&1 4>&2
+trap 'exec 2>&4 1>&3' 0 1 2 3
+exec 1> >(tee -a "${LOGFILE}") 2> >(tee -a "${LOGFILE}" >&2)
+
+BACKUPS_DIR="/var/tmp/postges_upgrade_backup"
```

Comment:
as this depends on the directory from the other script, the source of these should be the same, eg a pillar